core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -42,6 +42,16 @@ private SnapshotUtil() {
     return ancestorIds(table.currentSnapshot(), table::snapshot);
   }
 
+  /**
+   * @return List of snapshot ids in the range - (fromSnapshotId, toSnapshotId]
+   * This method assumes that fromSnapshotId is an ancestor of toSnapshotId
```

Comment:
I don't see any place that checks whether the from snapshot is an ancestor of the to snapshot. That seems like a requirement for this to work correctly to me.

Reply:
Yes. It will be in the follow-up